adafruit_max7219/bcddigits.py

```diff
@@ -9,6 +9,13 @@
 from micropython import const
 from adafruit_max7219 import max7219
 
+try:
+    import digitalio
+    import busio
+    from typing import List
+except ImportError:
```

Comment:
Thank you for working on this @FlantasticDan 

I think we want to do these imports in a slightly different order, the `typing` import first, and include a comment to explain what they're for.
```py
try:
    # Used only for typing
    from typing import List
    import digitalio
    import busio
except ImportError:
    pass
```
This way when it runs on a microcontroller it will fail to import typing, catch the exception and not import `digitalio` and `busio` which should save a bit of memory.

Lets also add a comment to the imports indicating they are only used for typing so that it's clear why they are in the try/except block.

Reply:
> This way when it runs on a microcontroller it will fail to import typing, catch the exception and not import digitalio and busio which should save a bit of memory.

This is a great point, makes me wonder if we want to add an un-necessary `typing` import to `adafruit_max7219/max7219.py` and `adafruit_max7219/matrices.py` just so that the import fails before importing `busio` and/or `digitalio` on microcontrollers.

For example, change `adafruit_max7219/max7219.py` to this even though that `typing` import will go unused:
```python
try:
    # Used only for typing
    import typing
    import busio
except ImportError:
    pass
```
What do you think @FoamyGuy ?

Reply:
Yep, good thought. That makes sense to me